### PR TITLE
 Profile Setup  #40012

### PR DESCRIPTION
**Pull request by Roshan-successive**
- Create a JWT token (by using id and email).
- Learned about model setups and seed.
- Model setup with initial seed.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,2 +1,3 @@
 PORT=9000
-NODE_ENV=dev
\ No newline at end of file
+NODE_ENV=dev
+MONGO_URL=mongodb://localhost:27017/express-training
```
Comment (suraj-bha) [inline .env]: This file needs to be added to gitignore.

File: src/repositories/user/IUserModel.ts [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+    email: string;
+    role: string;
+    password: string;
+=======
 id: string;
 name: string;
 role: string;
 password: string;
+>>>>>>> 621498a891e1574640081f10acfa1e01bf2bf2a8
 }
```
Comment (suraj-bha) [inline src/repositories/user/IUserModel.ts]: Need to merge conflicts correctly.

File: src/repositories/user/UserModel.ts [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+
+export const userModel: mongoose.Model<IUserModel> = mongoose.model<IUserModel>
+    (
+        'User',
+        userSchema,
+        'User',
+        true,
+    );
+=======
 import UserSchema from './UserSchema';
```
Comment (suraj-bha) [inline src/repositories/user/UserModel.ts]: conflicts need to be merged properly.

File: src/repositories/user/UserSchema.ts [hunk 4]
```diff
@@ -2,6 +2,21 @@ import * as mongoose from 'mongoose';
...
+            email: String,
+            role: String,
+            password: String
+        });
+        super(baseSchema, collection);
+    }
+}
+
+export default UserSchema;
+=======
```
Comment (suraj-bha) [inline src/repositories/user/UserSchema.ts]: Conflict needs to be merged properly.

**Review by suraj-bha — CHANGES_REQUESTED** [review]

File: src/repositories/user/UserSchema.ts (hunk 4 above)
Comment (Roshan-successive) [reply src/repositories/user/UserSchema.ts]: Resolved the conflicts

File: src/repositories/user/UserModel.ts (hunk 3 above)
Comment (Roshan-successive) [reply src/repositories/user/UserModel.ts]: Resolved the conflicts

File: .env (hunk 1 above)
Comment (Roshan-successive) [reply .env]: .env file added to gitignore

**Review by suraj-bha — APPROVED** [review]

**Pull request merged**